### ISSUE TITLE
Type error with getitem of StrEnum failed with 1.1335

### ISSUE THREAD

**Issue by jenshnielsen** [microsoft/pyright#6387]
**Describe the bug**
Type checking the following code using the python 3.11 StrEnum fails with pyright 1.1.335 but passed with 1.1.334 and earlier.

Typecheck fails with:

```
error: Expected 1 more positional argument (reportGeneralTypeIssues)
```

**Code or Screenshots**

```python
from enum import StrEnum

class MyEnum(StrEnum):
    A = "A"

MyEnum["A"]
```

**VS Code extension or command-line**
Commandline 1.1.335 on windows 11 but also seen in vscode using latest pylance prerelease. Python 3.11

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. I'm surprised that we didn't have this case covered already in pyright's test suite. Now it's covered, so this regression shouldn't happen again.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
